packages/wrangler/src/d1/execute.tsx

```diff
@@ -552,17 +555,26 @@ async function d1ApiPost<T>(
 	action: string,
 	body: unknown
 ) {
-	return await fetchResult<T>(
-		`/accounts/${accountId}/d1/database/${db.uuid}/${action}`,
-		{
-			method: "POST",
-			headers: {
-				"Content-Type": "application/json",
-				...(db.internal_env ? { "x-d1-internal-env": db.internal_env } : {}),
-			},
-			body: JSON.stringify(body),
+	try {
+		return await fetchResult<T>(
+			`/accounts/${accountId}/d1/database/${db.uuid}/${action}`,
+			{
+				method: "POST",
+				headers: {
+					"Content-Type": "application/json",
+					...(db.internal_env ? { "x-d1-internal-env": db.internal_env } : {}),
+				},
+				body: JSON.stringify(body),
+			}
+		);
+	} catch (x) {
+		if (x instanceof APIError) {
```

Comment:
fwiw the API endpoints now correctly throw a `400 Bad Request` when a user error occurs - might be worth specifically looking for that type of response

Reply:
Unfortunately, the way the cfetch code works it drops the HTTP status code by the time it generates an `APIError`. All we get back is:

```
  "result": [],
  "success": false,
  "errors": [
    {
      "code": 7500,
      "message": "no such table: Customers"
    }
  ],
  "messages": []
}
```

When creating the APIError, we do grab the error code from the first error in the response. In the example above 7500.

We could consider fixing the cfetch to throw an APIError whenever we don't get a 2xx response but I worry that this might break other APIs.